shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/data/repository/FeedingPointRepositoryImpl.kt

```diff
@@ -1,57 +1,91 @@
-            ),
-            lastFeeder = Feeder(
-                id = index,
-                name = "$index - Giorgi Abutidze",
-                time = "${Random.nextInt(0..24)} hours ago"
-            )
-        )
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+import com.amplifyframework.datastore.generated.model.FeedingPoint as DataFeedingPoint
+import com.epmedu.animeal.feeding.domain.model.FeedingPoint as DomainFeedingPoint
+
+internal class FeedingPointRepositoryImpl(
+    private val authAPI: AuthAPI,
+    private val favouriteApi: FavouriteApi,
+    private val feedingPointApi: FeedingPointApi
+) : FeedingPointRepository {
+
+    private val feedingPoints: Flow<List<DomainFeedingPoint>> by lazy { fetchFeedingPoints() }
+
+    private fun fetchFeedingPoints(): Flow<List<DomainFeedingPoint>> {
+        return feedingPointApi.getAllFeedingPoints()
+            .combine(favouriteApi.getFavouriteList(authAPI.currentUserId)) { dataFeedingPointsList, favouriteList ->
+                dataFeedingPointsList.map { dataFeedingPoint ->
+                    dataFeedingPoint.toDomainFeedingPoint(
+                        isFavourite = favouriteList.any { it.feedingPointId == dataFeedingPoint.id }
+                    )
+                }
+            }
     }
 
-    override fun getAllFeedingPoints(): Flow<List<FeedingPoint>> = flowOf(stubData)
+    override fun getAllFeedingPoints(): Flow<List<DomainFeedingPoint>> = feedingPoints
 
-    override fun getCats(): Flow<List<FeedingPoint>> = flowOf(
-        stubData.filter { feedingPoint -> feedingPoint.animalType == AnimalType.Cats }
-    )
+    override fun getCats(): Flow<List<DomainFeedingPoint>> {
+        return feedingPoints.map { feedingPointsList ->
+            feedingPointsList.filter { feedingPoint -> feedingPoint.animalType == AnimalType.Cats }
+        }
+    }
 
-    override fun getDogs(): Flow<List<FeedingPoint>> = flowOf(
-        stubData.filter { feedingPoint -> feedingPoint.animalType == AnimalType.Dogs }
-    )
+    override fun getDogs(): Flow<List<DomainFeedingPoint>> {
+        return feedingPoints.map { feedingPointsList ->
+            feedingPointsList.filter { feedingPoint -> feedingPoint.animalType == AnimalType.Dogs }
+        }
+    }
+
+    override fun getFavourites(): Flow<List<DomainFeedingPoint>> {
+        return feedingPoints.map { feedingPointsList ->
+            feedingPointsList.filter { feedingPoint -> feedingPoint.isFavourite }
+        }
+    }
 
-    override fun getFavourites(): Flow<List<FeedingPoint>> = flowOf(
-        stubData.filter { feedingPoint -> feedingPoint.isFavourite }
+    override fun getFeedingPoint(id: String): Flow<DomainFeedingPoint?> {
+        return feedingPoints.map { feedingPointsList ->
+            feedingPointsList.find { feedingPoint -> feedingPoint.id == id }
+        }
+    }
+
+    private fun DataFeedingPoint.toDomainFeedingPoint(isFavourite: Boolean) = DomainFeedingPoint(
```

Comment:
Would be better to keep mapping logic in a separate file

Reply:
Done